src/available-command.ts

```diff
@@ -329,7 +334,7 @@ export function possibleBoardActions(engine: Engine, player: Player) {
 
 }
 
-export function possibleFreeActions(engine: Engine, player: Player) {
+export function possibleFreeActions(engine: Engine, player: Player, gaiaPhase?: boolean) {
```

Comment:
I think the gaia phase indication should be in engine. Like there's already round in engine, there should also be the phase (income & gaia). That way, SetupFaction, SetupBuilding, SetupRoundBooster can also be added to phase, and round 0 can encompass all of these instead of having round -3. Maybe even leech can be a phase.

This will help with command management (deciding whether or not to combine two commands in the same line in the UI), and if things become more complicated it can also help with ordering subcommands.

Reply:
I agree, it would be much cleaner. I can work on this.Is it better to merge this PR and have another branch for phase?

Reply:
What if we use a FSM? https://github.com/eonarheim/TypeState?

Reply:
I don't think a FSM library is needed. And it will restrict changes that we can make to the code, and mechanisms that we use. (I'm still thinking about the future brainstone command)

I can merge this PR first, it just needs the two comments about brainstone addressed.